src/experimental/coroutines.rs

```diff
@@ -93,32 +136,35 @@ impl Coroutine {
         let context = &mut get_context().coroutines_context;
 
         let coroutine = &mut context.coroutines[self.id];
-        if let Some(f) = coroutine {
+        if let CoroutineState::Running(f) = coroutine {
             context.active_coroutine_now = f.manual_time;
             context.active_coroutine_delta = Some(delta_time);
             *f.manual_time.as_mut().unwrap() += delta_time;
-            if resume(&mut f.future) {
-                *coroutine = None;
+            if let Some(v) = resume(&mut f.future) {
+                *coroutine = CoroutineState::Value(v);
```

Comment:
With current implementation it is not really important if stopped/finished coroutine have `CoroutineState::Value(())` or `CoroutineState::Nothing` value, so this should not be addressed right now

but in the future, when coruitne's id's will be reused to stop leaking memory - it would be nice to somehow automatically auto-retrieve values of finished `()` coroutines and therefore change the state from `::Value(())` to `Nothing`

Reply:
I thought about adding a bool in `CoroutineInternal` that gets set when all `Coroutine` copies get dropped (using arc), it would require removing `Copy` from `Coroutine`.